### PR TITLE
Add Linux hardware tests; fix sensors fan/voltage discovery bug

### PR DESCRIPTION
**Pull request by dbwiddis**
Adds tests for Linux hardware classes using `@TempDir` sysfs fixtures, and fixes a latent bug discovered during test development.

## New Tests

- **LinuxGpuStatsTest** — Comprehensive coverage of amdgpu, i915, xe drivers, NVML paths, session lifecycle, and edge cases using stub subclasses and temp directories. Coverage: 0% → ~99% instruction, ~80% branch.
- **LinuxGraphicsCardTest** — Tests for `Attrs`, constructor/getters, `findDrmInfo` (PCI slot matching, fallback, no driver), `readUeventValue`, and `readDriverName` using temp directories with symlinks.
- **LinuxPowerSourceTest** — Removed `@EnabledOnOs(OS.LINUX)` (``buildPowerSource`` is pure logic). Added copy-constructor test and ``getPowerSources(path)`` tests covering uevent parsing, `PropByName.MAP`, adapter filtering (ADP/AC/USBC), `PRESENT=0` skip, and nonexistent/empty directories.
- **LinuxSensorsTest** — Tests for hwmon temperature discovery (priority selection, temp1 direct read, temp2-6 averaging), thermal_zone fallback and priority, fan speed reading, voltage reading, isPi path, and no-sensors-found.

## Production Code Changes

- **LinuxPowerSource**: Extract ``getPowerSources(String powerSupplyPath)`` package-private overload; public no-arg method delegates with ``SysPath.POWER_SUPPLY``.
- **LinuxGraphicsCard**: Extract ``findDrmInfo(String pciSlot, String drmPath)`` package-private overload; make ``readUeventValue`` and ``readDriverName`` package-private.
- **LinuxSensors**: Replace static final path constants and `IS_PI` with instance fields set via a new package-private constructor accepting ``hwmonBasePath``, ``thermalZoneBasePath``, and ``isPi``. No-arg constructor delegates with real values.

## Bug Fix

**LinuxSensors fan/voltage discovery was broken.** Inside ``populateSensorsMapFromHwmon()``, the fan and voltage scan called ``getSensorFilesFromPath(path, ...)`` where ``path`` was the resolved directory (e.g. ``hwmon0``). Since ``getSensorFilesFromPath`` appends ``0``, ``1``, etc. to iterate, it looked for ``hwmon00``, ``hwmon01`` — which never exist. Moved the fan/voltage scan outside the while loop and passed the base ``hwmonPath`` instead.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed Linux sensor fan and voltage discovery logic to correctly locate sensor files.

* **Improvements**
  * Made Linux graphics and power-source discovery more configurable to improve detection across environments.

* **Tests**
  * Added extensive tests for Linux graphics, GPU stats, power sources, and sensor discovery to increase reliability.

### CODE REVIEW

File: oshi-common/src/test/java/oshi/hardware/common/platform/linux/LinuxSensorsTest.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    }
+
+    @Test
+    void testIsPiReturnsDefaults() {
+        // isPi=true but vcgencmd won't work on test machine, so returns 0/empty
+        LinuxSensors sensors = new LinuxSensors("/nonexistent/hwmon", "/nonexistent/thermal_zone", true);
+        assertThat(sensors.queryCpuTemperature(), closeTo(0d, EPS));
+        assertThat(sensors.queryFanSpeeds().length, is(0));
+        assertThat(sensors.queryCpuVoltage(), closeTo(0d, EPS));
+    }
```
[the remark this answers is not in the document]
Comment (dbwiddis) [reply oshi-common/src/test/java/oshi/hardware/common/platform/linux/LinuxSensorsTest.java:40]: Looking at the code: when isPi=true, queryCpuTemperature() calls queryCpuTemperatureFromVcGenCmd() which runs ExecutingCommand.getFirstAnswer("vcgencmd measure_temp"). On any non-Pi system this returns empty string, so the method returns 0. The risk of vcgencmd being present on a CI runner is essentially zero — it's a Raspberry Pi-specific binary. I'll skip this one as over-engineering
